renku_notebooks/api/notebooks.py

```diff
@@ -222,7 +222,13 @@ def server_logs(user, server_name):
     server = get_user_server(user, server_name)
     if server:
         pod_name = server.get("state", {}).get("pod_name", "")
-        logs = read_namespaced_pod_log(pod_name)
+        try:
+            logs = read_namespaced_pod_log(pod_name)
+        # catch predictable k8s api errors and return a significative string
+        except Exception as e:
+            if hasattr(e, "body"):
+                k8s_error = json.loads(e.body)
+                logs = f"Logs unavailable: {k8s_error['message']}"
```

Comment:
You should set a default value for `logs` before the if statement; otherwise, you'll get an undefined name error when the if condition is false:
```
logs = ""
if hasattr ...
```

Reply:
good point! I initialized the variable as an empty string as suggested